packages/core/src/declareAtom.ts

```diff
@@ -136,13 +121,10 @@ export function declareAtom<State>(
     }
   }
 
-  reduce(_initAction, (state = initialState) => state)
+  reduce(_initAction, (state = initialState as TState) => state)
```

Comment:
Why not? `(state: TState = initialState) => state`

Reply:
here we need cast because initialState argument is union